docs/samples/Microsoft.ML.Samples/Dynamic/ConcatTransform.cs

```diff
@@ -0,0 +1,66 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+        // the alignment of the usings with the methods is intentional so they can display on the same level in the docs site.
+        using Microsoft.ML.Runtime.Data;
+        using Microsoft.ML.Runtime.Api;
+        using Microsoft.ML.Data;
+        using System;
+        using System.Linq;
+        using System.Collections.Generic;
+
+namespace Microsoft.ML.Samples.Dynamic
+{
+    public partial class TransformSamples
+    {
+        class SampleInfertDataWithFeatures
+        {
+            public VBuffer<int> Features { get; set; }
+        }
+
+        public static void ConcatTransform()
+        {
+            // Create a new ML context, for ML.NET operations. It can be used for exception tracking and logging, 
+            // as well as the source of randomness.
+            var ml = new MLContext(seed: 1, conc: 1);
+
+            // Get a small dataset as an IEnumerable.
+            IEnumerable<SamplesUtils.DatasetUtils.SampleInfertData> data = SamplesUtils.DatasetUtils.GetInfertData();
+            var trainData = ml.CreateStreamingDataView(data);
+
+            // Preview of the data.
+            // Age    Case  Education  induced     parity  pooled.stratum  row_num  ...
+            // 26.0   1.0   0-5yrs      1.0         6.0       3.0      1.0  ...
+            // 42.0   1.0   0-5yrs      1.0         1.0       1.0      2.0  ...
+            // 39.0   1.0   0-5yrs      2.0         6.0       4.0      3.0  ...
+            // 34.0   1.0   0-5yrs      2.0         4.0       2.0      4.0  ...
+            // 35.0   1.0   6-11yrs     1.0         3.0       32.0     5.0  ...
+
+            // A pipeline for concatenating the age, parity and induced columns together in the Features column.
+            string outputColumnName = "Features";
+            var pipeline = new ConcatEstimator(ml, outputColumnName, new[] { "Age", "Parity", "Induced"});
+
+            // The transformed data.
+            var transformedData = pipeline.Fit(trainData).Transform(trainData);
+
+            // Getting the data of the newly created column as an IEnumerable of SampleInfertDataWithFeatures.
+            var featuresColumn = transformedData.AsEnumerable<SampleInfertDataWithFeatures>(ml, reuseRowObject: false);
```

Comment:
What does `reuseRowObject` do? Would defaults be fine for these samples?

Reply:
for the enumerable, it determines whether to return the same object on every row, or allocate a new one per row. It is a required param; doesn't have a default. 

For the settings of the transforms, i am using both defaults and non-defaults; since the purpose of this snippet is to educate about usage. 

---
In reply to: [227591855](https://github.com/dotnet/machinelearning/pull/1321#discussion_r227591855) [](ancestors = 227591855)